XIPs/xip-reply-content-type.md

```diff
@@ -0,0 +1,61 @@
+---
+xip: 
+title: Reply Content Type
+description: Content type for reply messages
+author: Kunal Mondal
+status: Draft 
+type: Standards Track
+category: XRC
+created: 2023-03-24
+---
+
+## Abstract
+
+This XRC proposes a new content type for messages that enriches the messaging experience. It attempts to include only the bare minimum for clients to determine how to display such messages in order to provide flexibility for more rich types in the future.
+
+## Motivation
+
+When it comes to having a comfortable conversation, replying to messages is an absolute no brainer as it gives users the freedom to converse without any miscommunications. In order to provide users a real messaging platform, XMTP needs to provide this capability.
+
+## Specification
+
+Proposed content type:
+
+```js
+{
+  authorityId: "xmtp.org"
+  typeId: "reply"
+  versionMajor: 0
+  versionMinor: 1
+}
+```
+
+The reply message MUST include the following parameters:
+
+Adding info about the message we’re replying to (it’s content type id, content, sender address and the message id). 
+
+```ts
+{
+  replyContentTypeId: string
+  messageContentTypeId: string
+  replyContent: string
+  replyMessageId: string
+  replySenderAddress: string
+  messageContent: string
```

Comment:
What if the content type of the message you're replying to isn't text (like a `RemoteAttachment`), or your reply isn't text?

Also what if the `messageContentTypeId` doesn't match what the original message's was?

I think if possible we should just work with IDs here. Something like:

```ts
inReplyToID: string
```

where `inReplyToID` is the message ID you're replying to. The actual `content` could be a specific set of allowed content types like `Text` or `RemoteAttachment`?